File: C++/src/Solution2901_3000/Solution2908.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution2908.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int minimumSum(vector<int>& nums) {
  int n = nums.size();
  int res = INT_MAX;
  for (int j = 1; j < n - 1; ++j) {
    for (int i = 0; i < j; ++i) {
      if (nums[i] < nums[j]){
        for (int k = j + 1; k < n; ++k) {
          if (nums[j] > nums[k]){
            res = min(res, nums[i] + nums[j] + nums[k]);
          }
        }
      }
    }
  }
  return res == INT_MAX ? -1 : res;
}

File: Java/src/main/java/QuestionBank/Solution2901_3000/Solution2908.java
package QuestionBank.Solution2901_3000;
/**
 * @author windows
 */
public class Solution2908 {

    public int minimumSum(int[] nums) {
        int n = nums.length;
        int res = Integer.MAX_VALUE;
        for (int j = 0; j < n - 1; j++) {
            for (int i = 0; i < j; i++) {
                if (nums[j] > nums[i]) {
                    for (int k = j + 1; k < n; k++) {
                        if (nums[j] > nums[k]){
                            res = Math.min(res,nums[i] + nums[j] + nums[k]);
                        }
;                    }
                }
            }
        }
        return res == Integer.MAX_VALUE ? -1 : res;
    }
    
    
    public static void main(String[] args) {
        Solution2908 solution2908 = new Solution2908();
    }
}